var/spack/repos/builtin/packages/py-sqlalchemy/package.py

```diff
@@ -24,22 +25,22 @@ class PySqlalchemy(PythonPackage):
     version("1.1.18", sha256="8b0ec71af9291191ba83a91c03d157b19ab3e7119e27da97932a4773a3f664a9")
     version("1.0.12", sha256="6679e20eae780b67ba136a4a76f83bb264debaac2542beefe02069d0206518d1")
 
-    depends_on("python@2.7:2.8,3.6:", when="@1.4:", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.4:", when="@:1.3", type=("build", "run"))
-    depends_on("py-setuptools", type="build")
-    depends_on("py-importlib-metadata", when="@1.4.0: ^python@:3.7", type="run")
-    depends_on("py-greenlet@:0.4.16,0.4.18:", when="@1.4.0:", type="run")
-
     variant(
         "backend",
         description="Python modules for database access",
-        values=any_combination_of("mysql", "pymysql", "postgresql"),
+        values=any_combination_of("mysql", "postgresql", "pymysql"),
     )
 
+    depends_on("py-setuptools@47:", when="@2:", type="build")
+    depends_on("py-setuptools", type="build")
+    depends_on("py-importlib-metadata", when="@1.4.0: ^python@:3.7", type="run")
+    conflicts("^py-greenlet@0.4.17", when="@1.4.0:")
+    depends_on("py-typing-extensions@4.2.0", when="@2:", type="run")
```

Comment:
I would make these build/run so that `--test=root` passes. I'm hoping we can make these run-only someday but don't see an opportunity to do that any time soon (@tldahlgren)

Reply:
Didn't see that the line I copied from had the wrong type. Thanks.